lighthouse-core/test/computed/computed-artifact-test.js

```diff
@@ -38,4 +38,31 @@ describe('ComputedArtifact base class', () => {
     assert.equal(result, 1);
     assert.equal(computeCounter, 2);
   });
+
+  it('caches by strict equality on key list if provided', async () => {
+    let computeCounter = 0;
+    class RawTestComputedArtifact {
+      static async compute_() {
+        return computeCounter++;
+      }
```

Comment:
maybe

```suggestion
      static async compute_(dependencies) {
        assert.deepEqual(Object.keys(dependencies), ['x']);
        return computeCounter++;
      }
```
(and maybe pull out the `['x']` for here and the `makeComputedArtifact` call if it feels good)

to ensure the filtering is working?